DQMOffline/Trigger/plugins/HTMonitor.cc

```diff
@@ -201,36 +229,57 @@ void HTMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup)
   if (!jets.empty()) deltaPhi_met_j1 = fabs( deltaPhi( pfmet.phi(),  jets[0].phi() ));
   if (jets.size() >= 2) deltaPhi_j1_j2 = fabs( deltaPhi( jets[0].phi(),  jets[1].phi() ));
 
+  std::vector<reco::GsfElectron> electrons;
   edm::Handle<reco::GsfElectronCollection> eleHandle;
   iEvent.getByToken( eleToken_, eleHandle );
-  std::vector<reco::GsfElectron> electrons;
-  if ( eleHandle->size() < nelectrons_ ) return;
-  for ( auto const & e : *eleHandle ) {
-    if ( eleSelection_( e ) ) electrons.push_back(e);
+  if ( eleHandle.isValid() ) {
+    if ( eleHandle->size() < nelectrons_ ) return;
+    for ( auto const & e : *eleHandle ) {
+      if ( eleSelection_( e ) ) electrons.push_back(e);
+    }
+    if ( electrons.size() < nelectrons_ ) return;
+  } else {
+    if (!warningPrinted4token_[2]) {
+      edm::LogWarning("HTMonitor") << "skipping events because the collection " << jetInputTag_.label().c_str() << " is not available";
```

Comment:
hi-
I generally find these changes distasteful - however one practical comment - there is nothing here that actually skips the event. instead any checks on the number of good electrons are just skipped Probably that is not the behavior desired. 
(note also that the variable electrons is completely unnecessary it seems)

Reply:
@davidlange6 as discussed directly we agree, I consider this as a quick workaround, the final solution is to properly add the producer in the correct sequence

Reply:
if the input electron collection is wrong, because not available, I doubt one can rely on any outcome of this code
=> cutting or not on the electrons multiplicity loses its value, I think
said that, I'm almost sure not all the consumers / users of this module need electrons,
=> I preferred to keep a conservative approach, which allows to process the event

Reply:
@mtosi but you would need a return statement after setting the flag to true, right?

Reply:
the selections on electron will not be done, but the plots will be (wrongly) filled ...
but, if one does not want to apply any cuts on the electrons multiplicity and indeed even misconfigured the module, the plot will be (properly) filled and fine, and the warning message will make people know to update the config

am I wrong ?